### DISCUSSION TITLE
[HELP] Connecting to SPH6000

### DISCUSSION THREAD

**Discussion by exico91** [wills106/homeassistant-solax-modbus#498] [Q&A]
**Describe the bug**
Growatt SPH6000 no data

**Mandatory details**
1. Integration Version:  2023.06.8
2. HA Version: 2023.07.1
3. Inverter Brand: Growatt
4. Plugin Used: growatt
5. First 6 digits of Serial: NVCRBK
6. Connection Method: RS485 -> USB

**Detailed Error Log**
[home-assistant_2023-07-17T13-47-55.589Z.log](https://github.com/wills106/homeassistant-solax-modbus/files/12069984/home-assistant_2023-07-17T13-47-55.589Z.log)

**Additional context**
Since i did not found a "how to install" but only some notes I puzzled together what i found:
- I installed from HACS the addon and restarted HA
- I connected a RJ45 cable to the RS485 port on the inverter and i used this https://www.waveshare.com/usb-to-rs485-b.htm to connect it to homeassistant using the blue pair, Blue on A and Blue striped on B
- I searched for the serial adapter in HA, (/dev/serial/by-id/usb-1a86_USB_Single_Serial_5185024413-if00)
- I added the plugin via HA with these settings: serial, address 1, inverter type growatt, 15 seconds, serial port name: /dev/serial/by-id/usb-1a86_USB_Single_Serial_5185024413-if00, baud 9600

Im doing something wrong? I have to configure the modbus in the configuration.yaml? If yes with wich settings?

**Comment by wills106** [comment]
You don't need to configure anything in connfiguration.yaml

Are you sure you are using a baud of 9600 as that log shows 115200

I would double check the manual for the Growatt Inverter to check which pins the RS485 connection is on. Have you looked in the Inverter LCD to see if Modbus is actually enabled? I know on some SolaX you need to enable it in a sub menu.

**Comment by exico91** [comment]
I tried different baud rates to see if something came up to no avail. I checked a manual and it says 9600 so i left it there after the tries.

I connected a RJ45 to the RS485 port on the inverter
There is a COM port with a wire push-in type connector but i do not know the pinout and Im not sure if its the correct port. (i attached a couple photos)

I checked the menu (had to press OK for 5 seconds to bring it up), there is a RS485 menu and it lets me select Battery or VPP, It was on Battery and i changed it to VPP but nothing came up.

I also tried a modbus scanner with the serial adapter and nothing came up in the bus so I guess that the inverter is not responding or has the RS485 disabled for some reason.

![Immagine WhatsApp 2023-07-17 ore 11 26 03](https://github.com/wills106/homeassistant-solax-modbus/assets/97479177/81ca24c3-b6f7-48f3-9870-6ee2c4e866ef)
![Immagine WhatsApp 2023-07-17 ore 12 44 54](https://github.com/wills106/homeassistant-solax-modbus/assets/97479177/3db7b639-d978-428b-ae11-aab44275d2e2)
![Immagine WhatsApp 2023-07-17 ore 12 42 56](https://github.com/wills106/homeassistant-solax-modbus/assets/97479177/92d9fdff-79e9-4e15-8d32-9b45fa4f4201)

If we can make this to work I can also write a guide to add to the wiki

**Comment by exico91** [comment]
I found this Info page and seems that this inverter model has different RS485 Pins, I though that they were the blue pair on a ethernet cable (4 & 5) but according to this page seems that 5 is A and 1 is B on the RS485 

https://solar-assistant.io/help/growatt/sph-range

Will try on the first occasion

**Comment by wills106** [comment]
Yea the same brand Inverter can have the RS485 connection on different pins on different models.
Always best to double check the documentation.

Goodluck.

**Comment by exico91** [comment]
Unfortunately i cannot get any data out of this inverter. Ive tried with different pins but nothing.

**Comment by wills106** [comment]
Moved to Discussions as not an Integration issue.

**Comment by danmiles86** [comment]
Was anyone able to confirm the pinout? I'm having the same issue but I suspect its just incorrect pinout.

**Reply by Raageri** [reply]
I have 8 ports in my WIT 8K- HU. Port marked meter didn't work but "normal" 485 did labelled as 485_1. It was Blue and blue white wire which enabled communication. From port "meter" it didnt work. 

<img width="1111" height="279" alt="image" src="https://github.com/user-attachments/assets/496fc9a5-b734-4cfa-bf2f-59479b44bb25" />
